### PR TITLE
chore: Migrating from node-riffraff-artifact to actions-riff-raff

### PR DESCRIPTION
**Pull request by akinsola-guardian**
## What does this change?
Migrating from node-riffraff-artifact to actions-riff-raff
## Why?
node-riffraff-artifact has been deprecated.

Tested in CODE

Trello: https://trello.com/c/AxnZQXCM

### CODE REVIEW

File: cdk/script/export-monitoring-file [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+then
+	src_file=$(find $INDEX_JS_BUILT_FILE) # Gets the index.js file generated in the monitoring/dist. Throws error if it doesn't exist
+	absolute_src_file_path=$(get_abs_filename "$src_file") # Gets the absolute path of the index.js file
+	absolute_src_directory=$(dirname "$absolute_src_file_path") # Gets the src directory name the absolute_src_file_path
+	echo "----------------------------"
+
+	for region in "${LAMBDA_REGIONS[@]}"
+	do
+		echo "starting ${region}"
+		cd "${absolute_src_directory}" # Go to monitoring/dist folder
```
Comment (AshCorr) [inline cdk/script/export-monitoring-file:25]:
```suggestion
		cd "./monitoring/dist" # Go to monitoring/dist folder
```

Feels like a lot of code is dedicated to finding absolute paths, wondering why not just use relative paths?

**Review by AshCorr — APPROVED** [review]
Looks good!

Just a couple of comments about potentially simplifying!

File: .github/workflows/monitoring-CI.yml [hunk 2]
```diff
@@ -65,5 +65,34 @@ jobs:
...
+          configPath: riff-raff.yaml
+          contentDirectories: |
+            cmp-monitoring-lambda-eu-west-1:
+              - cdk/dist/cmp-monitoring-lambda-eu-west-1.zip
+            cmp-monitoring-lambda-us-west-1:
+              - cdk/dist/cmp-monitoring-lambda-us-west-1.zip
+            cmp-monitoring-lambda-ap-southeast-2:
+              - cdk/dist/cmp-monitoring-lambda-ap-southeast-2.zip
+            cmp-monitoring-lambda-ca-central-1:
+              - cdk/dist/cmp-monitoring-lambda-ca-central-1.zip
```
[the remark this answers is not in the document]
Comment (akinsola-guardian) [reply .github/workflows/monitoring-CI.yml]: Thank you, This make a lot of sense. I've just simplified this.

File: cdk/script/export-monitoring-file (hunk 1 above)
Comment (akinsola-guardian) [reply cdk/script/export-monitoring-file]: I tried to do this but didn't quite work in Github action. I've abstracted away into a function and added more comments to make it clearer.

File: cdk/script/export-monitoring-file [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+if test -f "${INDEX_JS_BUILT_FILE}"; # Check if index js file is in the monitoring file
+then
+	src_file=$(find $INDEX_JS_BUILT_FILE) # Gets the index.js file generated in the monitoring/dist. Throws error if it doesn't exist
+	absolute_src_directory=$(get_absolute_path_directory "$src_file") # Gets the directory of the index.js file monitoring/dist
+
+	echo "Zipping Built Files"
+	cd "${absolute_src_directory}" # Go to monitoring/dist folder
+	zip -FSr "${ROOT_DIR}/cdk/dist/cmp-monitoring-lambda.zip" "." # Zip the monitoring/dist folder into cdk/dist
+	echo "File was successfully zipped ${region}"
```
Comment (sookburt) [inline cdk/script/export-monitoring-file:23]: Is region still required here if we're no longer needing that?

Comment (akinsola-guardian) [reply cdk/script/export-monitoring-file]: Thanks for the spot.

**Review by sookburt — APPROVED** [review]
Deployed to test environment and tested in all 4 jurisdictions.  Looks great!

**Pull request merged**